### PR TITLE
feat(world): remove system name from function signatures/selectors [M-05]

### PR DESCRIPTION
**Pull request by holic**
closes https://github.com/latticexyz/mud/issues/1980

### CODE REVIEW

File: .changeset/bright-kangaroos-battle.md [hunk 1]
```diff
@@ -0,0 +1,8 @@
+---
+"@latticexyz/world-modules": patch
+"@latticexyz/world": major
+---
+
+Functions within namespaced systems are now registered on the world as `{namespace}_{functionName}` rather than `{namespace}_{systemName}_{functionName}`. This is more ergonomic and is more consistent with namespaced resources in other parts of the codebase (e.g. accessing tables in schemaful indexer).
+
+If you have a project using the `namespace` key in your `mud.config.ts` or are manually registering systems and function selectors on a namespace, you will likely need to codegen your system interfaces (`pnpm build`) and update any calls to these systems through the world's namespaced function signatures.
```
Comment (holic) [inline .changeset/bright-kangaroos-battle.md:8]: Is there a way to bubble this up as a breaking change aside from `major` tag?

**Comment by yonadaaa** [comment]
No suggestions yet but I wanna experiment with this a bit, think about what happens when you register multiple systems with the same name etc.

Maybe document everything in tests if not already covered

**Comment by holic** [comment]
> think about what happens when you register multiple systems with the same name

@alvrs and I chatted a bit about this IRL and we think it'll just end up with a compile error, because codegen will create interfaces for each system and those will get combined into a `IWorld` interface, but that won't compile if there are function signature/selector overlaps

**Comment by alvrs** [comment]
@qbzzt there might be a couple places in the docs we need to update with this!

File: .changeset/bright-kangaroos-battle.md (hunk 1 above)
Comment (alvrs) [reply .changeset/bright-kangaroos-battle.md:8]: major should be only breaking changes, no other way make it more prominent unfortunately

**Review by alvrs — DISMISSED** [review]

**Comment by yonadaaa** [comment]
>  it'll just end up with a compile error, because codegen will create interfaces for each system and those will get combined into a IWorld interface, but that won't compile if there are function signature/selector overlaps

The compiler error is unfortunate, I wonder if we can surface something more helpful in codegen? Trying it locally I get:

image

**Comment by holic** [comment]
> The compiler error is unfortunate, I wonder if we can surface something more helpful in codegen? Trying it locally I get:
> 
> image

Hmm, those don't appear to be namespaced? But the compiler does communicate well enough that there's an overlap.

We could def add more validation+errors to codegen.

**Comment by yonadaaa** [comment]
This makes me wonder if we can/should hide the concept of Systems, instead focusing on functions and having "one contract interface per namespace". 

Instead of `registerSystem`, it could be `registerFunction`. Each function is registered separately along with the address of it's bytecode.

Need to think on this more but maybe it would clear things up here. Definitely out of scope for this PR!

**Comment by yonadaaa** [comment]
> Hmm, those don't appear to be namespaced? 

Oh, oops, they weren't. And you're right that the compiler error is pretty decent.

**Review by yonadaaa — DISMISSED** [review]

**Review by alvrs — APPROVED** [review]

**Pull request merged**